toolchain/semantics/semantics_context.cpp

```diff
@@ -379,6 +379,44 @@ auto SemanticsContext::ImplicitAsImpl(SemanticsNodeId value_id,
     // Type doesn't need to change.
     return ImplicitAsKind::Identical;
   }
+
+  auto array_type = semantics_ir_->GetTypeAllowBuiltinTypes(as_type_id);
+  auto array_node = semantics_ir_->GetNode(array_type);
+  if (array_node.kind() == SemanticsNodeKind::ArrayType) {
+    auto [bound_node_id, element_type_id] = array_node.GetAsArrayType();
+    if (semantics_ir_
+            ->GetNode(semantics_ir_->GetTypeAllowBuiltinTypes(value_type_id))
+            .kind() == SemanticsNodeKind::TupleType) {
+      auto tuple_type_block_id =
+          semantics_ir_
+              ->GetNode(semantics_ir_->GetTypeAllowBuiltinTypes(value_type_id))
+              .GetAsTupleType();
+      auto type_block = semantics_ir_->GetTypeBlock(tuple_type_block_id);
```

Comment:
```suggestion
      const auto& type_block = semantics_ir_->GetTypeBlock(tuple_type_block_id);
```
Avoid making a copy of the type block.